packages/scaffold-config/src/detect.ts

```diff
@@ -131,11 +131,11 @@ export function detectLanguage (projectRoot: string, pkgJson: PkgJson): 'js' | '
   }
 
   const globs = [
-    joinPosix(projectRoot, '**/*tsconfig.json'),
-    joinPosix(projectRoot, 'cypress', '**/*.{ts,tsx}'),
+    joinPosix('**/*tsconfig.json'),
+    joinPosix('cypress', '**/*.{ts,tsx}'),
   ]
 
-  const tsFiles = globby.sync(globs, { onlyFiles: true, gitignore: true })
+  const tsFiles = globby.sync(globs, { onlyFiles: true, gitignore: true, cwd: projectRoot, ignore: ['node_modules'] })
```

Comment:
I think we can just use the `async` call of `globby` to avoid EMFILE errors. Unfortunately, making this function as `async` has a huge cascading effect requiring 10s of files to be changed. We end up calling a function (that now needs to be `awaited` in a few constructors, too, which makes it more painful, since a constructor cannot be `async`, the design and flow of some parts of the system would need to be reconsidered.

We should still make this change, it's just not as trivial as "use await, mark async".